spec/system/flaky_spec/retry_spec.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+# rubocop:disable RSpec/BeforeAfterAll
+
+RSpec.describe "Testing FlakySpec::Retry on system tests" do
+  def set_expectations(expectations)
+    @expectations = expectations
+  end
+
+  def shift_expectation
+    @expectations.shift
+  end
+
+  def reset_expectations
+    @expectations = nil
+  end
+
+  def count
+    @count ||= 0
+  end
+
+  def count_up
+    @count ||= 0
+    @count += 1
+  end
+
+  def reset_count
+    @count = 0
+  end
+
+  _previous_example = nil
+
+  context "when test fails 2 times before passing on third try", order: :defined do
+    before(:all) { set_expectations(["not ok", "not ok", "ok"]) }
+
+    before { count_up }
+
+    after(:all) do
+      reset_expectations
+      reset_count
+      _previous_example = nil
+    end
+
+    it "should retry the example 2 times", type: :system, flaky_spec_retry: true do
+      visit("/srv/status")
+
+      expect(page).to have_content(shift_expectation)
+      expect(count).to eq(3)
+
+      _previous_example = RSpec.current_example
+    end
+
+    it "should store the right failure details in metadata of the flaky example" do
```

Comment:
Aren't examples usually shuffled? How do you make sure that `_previous_example` is set in this spec? Or are only spec files shuffled?